blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -206,6 +212,20 @@ public Resource resolve(Item context, Reachable parent, Item target) {
     }
 
     @Override
+    public int getNumberOfRunningPipelines(){
+        RunList runningJobs =  job.getBuilds().filter(isRunning);
+        return Iterators.size(runningJobs.iterator());
+    }
+
+    @Override
+    public int getNumberOfQueuedPipelines() {
+        if(job instanceof BuildableItem) {
+            return Iterables.size(Jenkins.getInstance().getQueue().getItems((BuildableItem)job));
```

Comment:
I guess the real question is this is an `AbstractPipelineImpl` ? but it's getting all the `BuildableItems` in Jenkins? Should it be restricted just to items related to this pipeline? Otherwise should this API be elsewhere?